doc/development/release_distribution.rst

```diff
@@ -15,137 +15,149 @@ Preamble
 This document covers the process for managing updates to the current minor release and making new releases.
 Within this document, the git remote ``upstream`` refers to the main QuTiP organsiation repository, and ``origin`` refers to your personal fork.
 
-Instructions on how to backport bugfixes to a release branch are detailed in bugfix_.
-You need to do this to make changes to a current release that cannot wait until the next minor release, but need to go out in a micro release as soon as possible.
+In short, the steps you need to take are:
 
-Follow either release_ if you are making a major or minor release, or microrelease_ instead if it is only a bugfix patch to a current release.
-For both, then do the following steps in order:
+1. Prepare the relevant ``git`` branch with the correct state of the code (git_).
+2. Build the binary and source versions of the package, and deploy it to PyPI (``pip``) (deploy_).
+3. Retrive the built documentation from GitHub (also in docbuild_).
+4. Release the files on the QuTiP GitHub page (github_).
+5. Update `qutip.org <http://qutip.org/>`_ with the new links and documentation (web_).
+6. Update the conda feedstock, deploying the package to ``conda`` (cforge_).
 
-1. docbuild_, to build the documentation
-2. deploy_, to build the binary and source versions of the package, and deploy it to PyPI (``pip``)
-3. github_, to release the files on the QuTiP GitHub page
-4. web_, to update `qutip.org <http://qutip.org/>`_ with the new links and documentation
-5. cforge_, to update the conda feedstock, deploying the package to ``conda``
 
 
-.. _gitwf:
+.. _git:
 
-Git workflow
-++++++++++++
+Setting Up The Release Branch
++++++++++++++++++++++++++++++
 
-.. _bugfix:
-
-Apply bug fix to latest release
--------------------------------
-Assuming that the bug(s) has been fixed in some commit on the master,
-then this bug(s) fix should now be applied to the latest release.
-First checkout ``master``, use ``$ git log`` to list the commits,
-and copy the hash(es) for the bug fix commit(s) to some temporary file or similar.
-
-Now check out latest version branch, e.g.
-
-If you have checked out this branch previously, then ::
-
-    $ git checkout qutip-4.0.X
-    $ git pull upstream qutip-4.0.X
-
-Otherwise ::
-
-    $ git fetch upstream
-    $ git checkout -b qutip-4.0.X upstream/qutip-4.0.X
-    $ git push -u origin qutip-4.0.X
+In this step you will prepare a git branch on the main QuTiP repository that has the state of the code that is going to be released.
+This procedure is quite different if you are releasing a new minor or major version compared to if you are making a bugfix patch release.
```

Comment:
Is the procedure not the same except that if the `.X` branch doesn't exist you have to create it first?

Reply:
I see them as fairly different, aside from the changelog stuff, which I separated out to make it clear it's for both of them.  That said, it depends a little on the audience - to me and you I think "cherry-pick what you want onto a branch" is a very simple operation and not wildly dissimilar to "branch off and update a couple of files", but in the context of having to explain how to enumerate the commits and then how each one should be cherry-picked, it's really a very big difference between the two.   The files you need to update are different between the two as well, so there's a fair amount of difference.

Reply:
Having read a bit further, I'm happy with having the two split.